### PR TITLE
Improve RenderingServer, RenderingDevice, ShaderGlobalsOverride documentation

### PR DESCRIPTION
**Pull request by Calinou**
This brings the overall class reference completion percentage from 87% to 92%.

The journey was long, but it was [worth](https://github.com/godotengine/godot/pull/75908) [it](https://github.com/godotengine/godot/pull/76009).

### CODE REVIEW

File: doc/classes/RenderingDevice.xml [hunk 1]
```diff
@@ -560,6 +692,12 @@
 			<param index="8" name="dst_layer" type="int" />
 			<param index="9" name="post_barrier" type="int" enum="RenderingDevice.BarrierMask" default="7" />
 			<description>
+				Copies the [param from_texture] to [param to_texture] with the specified [param from_pos], [param to_pos] and [param size] coordinates. The Z axis of the [param from_pos], [param to_pos] and [param size] must be [code]0[/code] for 2-dimensional textures. Source and destination mipmaps/layers must also be specified, with these parameters being [code]0[/code] for textures without mipmaps or single-layer textures. Returns [constant @GlobalScope.OK] if the texture copy was successful or [constant @GlobalScope.ERR_INVALID_PARAMETER] otherwise.
+				[b]Note:[/b] [param from_texture] texture can't be copied while a render pass that uses it is being created. Ensure the render pass is finalized and that it was created with [constant RENDER_PASS_CONTENTS_FINISH] to unbind this texture.
```
Comment (Calinou) [inline doc/classes/RenderingDevice.xml:696]: `RENDER_PASS_CONTENTS_FINISH` is mentioned in several of RenderingDevice's own error messages, but it appears this constant was never committed to Godot.

@reduz Was this part of an early design? Which flag should be used now?

Comment (Calinou) [reply doc/classes/RenderingDevice.xml]: Fixed. I've also adjusted error messages accordingly: https://github.com/godotengine/godot/pull/76008/files#diff-0d547c4e6687e653e54df1623d33ab291396cca8c6265101c211a028e610b1e9

**Review by mhilbrunner — APPROVED** [review]

**Review by clayjohn — COMMENTED** [review]
I started reviewing this earlier but it looks like I forgot to submit my comments. Very sorry about that.

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thanks, and thanks Clay for the review! Merged.

**Comment by akien-mga** [comment]
Cherry-picked for 4.0.3.

**Pull request merged**